app/src/main/java/com/android/shelfLife/ui/recipes/Recipes.kt

```diff
@@ -73,107 +73,114 @@ fun RecipesScreen(navigationActions: NavigationActions) {
   val filteredRecipeList by recipesViewModel.filteredRecipes.collectAsState()
   val selectedFilters by recipesViewModel.selectedFilters.collectAsState()
 
-  if (selectedHousehold == null) {
-    FirstTimeWelcomeScreen(navigationActions, overviewScreenViewModel) // TODO("kill yourself")
-  } else {
-    if (user != null) {
-      Scaffold(
-          modifier = Modifier.testTag("recipesScreen"),
-          topBar = {
-            selectedHousehold.value?.let {
-              TopNavigationBar(
-                  houseHold = it,
-                  onHamburgerClick = {
-                    recipesViewModel.viewModelScope.launch {
-                      recipesViewModel.drawerState.value.open()
-                    }
-                  },
-                  filters = recipesViewModel.filters,
-                  selectedFilters = selectedFilters,
-                  onFilterChange = { filter, isSelected -> recipesViewModel.clickOnFilter(filter) })
-            }
-          },
-          bottomBar = {
-            BottomNavigationMenu(
-                onTabSelect = { destination -> navigationActions.navigateTo(destination) },
-                tabList = LIST_TOP_LEVEL_DESTINATION,
-                selectedItem = Route.RECIPES)
-          },
-          // Floating Action Button to add a new food item
-          floatingActionButton = {
-            Column(
-                horizontalAlignment = Alignment.End,
-                verticalArrangement = Arrangement.spacedBy(16.dp),
-                modifier = Modifier.padding(end = 16.dp, bottom = 16.dp)) {
-                  // Secondary FAB for "Manual" option
-                  if (recipesViewModel.fabExpanded.value) {
-                    ExtendedFloatingActionButton(
-                        text = { Text("Generate") },
-                        icon = { Icon(Icons.Default.AutoAwesome, contentDescription = "Add") },
-                        onClick = {
-                          // Navigate to Manual Recipe screen
-                          navigationActions.navigateTo(Screen.GENERATE_RECIPE)
-                          recipesViewModel.shrinkFab()
-                        },
-                        containerColor = MaterialTheme.colorScheme.secondaryContainer,
-                        modifier = Modifier.testTag("generateRecipeFab").width(150.dp))
-                  }
+  val drawerState by overviewScreenViewModel.drawerState.collectAsState()
+  val scope = rememberCoroutineScope()
 
-                  // Primary FAB
-                  ExtendedFloatingActionButton(
-                      text = { Text(if (recipesViewModel.fabExpanded.value) "Manual" else "") },
-                      icon = { Icon(Icons.Default.Add, contentDescription = "Add") },
-                      onClick = {
+  HouseHoldSelectionDrawer(
```

Comment:
## Good.
The changes look strange, but overall the functionality is kept the same